### ISSUE TITLE
Root screen lifecycle issue

### ISSUE THREAD

**Issue by guyca** [wix/react-native-navigation#2286]
It appears we have a lifecycle issue with root screen when reloading the app

Added some logs to `WelcomeScreen`

```js
class WelcomeScreen extends Component {
  componentDidMount() {
    console.log('guyca', `componentDidMount`);
  }

  componentWillUnmount() {
    console.log('guyca', `componentWillUnmount`);
  }
}
```

got the following output

```
12-05 16:32:23.861  3036  3664 I ReactNativeJS: ‘guyca’, ‘componentWillUnmount’
12-05 16:32:24.114  3036  3717 I ReactNativeJS: ‘guyca’, ‘componentDidMount’
12-05 16:32:24.147  3036  3717 I ReactNativeJS: ‘guyca’, ‘componentWillUnmount’
12-05 16:32:24.219  3036  3717 I ReactNativeJS: ‘guyca’, ‘componentDidMount’
```

**Comment by michael-gillett** [comment]
I am facing the same issue in 2.0.2083 on Android. I'm curious if this needs a similar fix to #2503, which fixed a similar issue, but with popping. Potentially the screens aren't being marked as destroyed on reload so they persist after refresh.

**Comment by guyca** [comment]
@michael-gillett Could definitely be related

**Comment by guyca** [comment]
Fixed in 096481e49602eb4fdcc33deadba92c2afb34d848
